src/private.rs

```diff
@@ -167,7 +167,11 @@ impl std::fmt::Debug for RedactionFormatter<'_> {
                     .and_then(|inner| inner.strip_suffix(')'))
                 {
                     fmt.write_str("Some(")?;
-                    self.flags.redact_partial(fmt, inner)?;
+                    if let RedactionLength::Partial = &self.flags.redact_length {
+                        self.flags.redact_partial(fmt, inner)?;
+                    } else {
+                        self.flags.redact_full(fmt, inner)?;
+                    }
```

Comment:
Here is the bug. If the value being redacted is of type `Option` then `redact_partial` is called no matter what, even if that option was assigned `#[redact]`